src/VisualStudio.Roslyn.SDK/ComponentDebugger/DebugProfileProvider.cs

```diff
@@ -0,0 +1,89 @@
+        {
+            _configuredProject = configuredProject;
+            _launchSettingsManager = launchSettingsManager;
+            _threadingService = threadingService;
+
+            _compilerRoot = new AsyncLazy<string>(() => GetCompilerRootAsync(serviceProvider), _threadingService.JoinableTaskFactory);
+        }
+
+        public Task OnAfterLaunchAsync(DebugLaunchOptions launchOptions, ILaunchProfile profile) => Task.CompletedTask;
+
+        public Task OnBeforeLaunchAsync(DebugLaunchOptions launchOptions, ILaunchProfile profile) => Task.CompletedTask;
+
+        public bool SupportsProfile(ILaunchProfile? profile) => Constants.CommandName.Equals(profile?.CommandName, StringComparison.Ordinal);
+
+        public async Task<IReadOnlyList<IDebugLaunchSettings>> QueryDebugTargetsAsync(DebugLaunchOptions launchOptions, ILaunchProfile? profile)
+        {
+            // set up the managed (net fx) debugger to start a process
+            // https://github.com/dotnet/roslyn-sdk/issues/729
+            var settings = new DebugLaunchSettings(launchOptions)
+            {
+                LaunchDebugEngineGuid = Microsoft.VisualStudio.ProjectSystem.Debug.DebuggerEngines.ManagedOnlyEngine,
+                LaunchOperation = DebugLaunchOperation.CreateProcess
+            };
+
+            // try and get the target project
+            var targetProjectUnconfigured = await _launchSettingsManager.TryGetProjectForLaunchAsync(profile);
+            if (targetProjectUnconfigured is object)
+            {
+                settings.CurrentDirectory = Path.GetDirectoryName(targetProjectUnconfigured.FullPath);
+                var compiler = _configuredProject.Capabilities.Contains(ProjectCapabilities.VB) ? "vbc.exe" : "csc.exe";
+                var compilerRoot = await _compilerRoot.GetValueAsync();
+                settings.Executable = Path.Combine(compilerRoot, compiler);
+
+                // get its compilation args
+                var args = await targetProjectUnconfigured.GetCompilationArgumentsAsync();
+
+                // append the command line args to the debugger launch
+                settings.Arguments = string.Join(" ", args);
+            }
+
+            // https://github.com/dotnet/roslyn-sdk/issues/728 : better error handling
+            return new IDebugLaunchSettings[] { settings };
+        }
+
+        private async Task<string> GetCompilerRootAsync(SVsServiceProvider? serviceProvider)
+        {
+            await _threadingService.SwitchToUIThread();
+
+            // https://github.com/dotnet/roslyn-sdk/issues/729 : don't hardcode net fx compiler
+            object rootDir = string.Empty;
+            var shell = (IVsShell?)serviceProvider?.GetService(typeof(SVsShell));
+            shell?.GetProperty((int)__VSSPROPID2.VSSPROPID_InstallRootDir, out rootDir);
+            return Path.Combine((string)rootDir, "MSBuild", "Current", "Bin", "Roslyn");
```

Comment:
Feel like the `?.` is hiding a bug here. If `shell is null` here then we end up calling `Path.Combine("", ...)` which will just return a relative path to `csc.exe` vs a full one. That's not going to execute the way we like. Think we should just proactively error here if `shell is null`